src/main/java/book/store/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package book.store.service.impl;
+
+import book.store.dto.CategoryDto;
+import book.store.mapper.CategoryMapper;
+import book.store.model.Category;
+import book.store.repository.CategoryRepository;
+import book.store.service.CategoryService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public CategoryDto save(CategoryDto categoryDto) {
+        Category category = categoryMapper.toModel(categoryDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream().map(categoryMapper::toDto).toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        return categoryMapper.toDto(categoryRepository.findById(id).get());
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryDto categoryDto) {
+        Category category = categoryMapper.toModel(categoryDto);
```

Comment:
First, find Category by id
Second, use MapStruct to update `mapper.updateCategoryFromDto(categoryDto, category);`